### ISSUE TITLE
Include "Already Registered" Message on Registration Form

### ISSUE THREAD

**Issue by spencerfinnell** [easydigitaldownloads/easy-digital-downloads#4922]
Much like: https://github.com/easydigitaldownloads/easy-digital-downloads/blob/c62bf93bd885d338b925e9e903ca33b4e0564d2d/templates/shortcode-login.php#L40

It's come up in our support more than a few times wondering why the registration page is blank.

**Comment by pippinsplugins** [comment]
I have doubts about this, simply because I've seen a _lot_ of tickets complaining about the opposite: message shown when already registered / logged in.

Probably the only way to please everyone is to support showing a message and supporting showing nothing.

**Comment by spencerfinnell** [comment]
What do the people who don't want the message want instead? A blank page or a redirect?

**Comment by pippinsplugins** [comment]
They want nothing. That allows them to combine registration, log in, and profile editors on a single page.

**Comment by lalunecreative** [comment]
I agree with the nothing but an option for both would be best. We run into a similar problem with our site because like most store owners they want a "account page" with everything on it. On ours we added downloads, points, and history along with the edd login form. Ideally, that page would show the login information when you're not logged in and when you are, not show anything. There's no point to telling you that you're logged in since you know already. I actually wish there was more support/features for an overall my account page.

**Comment by pippinsplugins** [comment]
Not an exact request for this but similar: https://secure.helpscout.net/conversation/249856600/42067/?folderId=180504

**Comment by spencerfinnell** [comment]
I can definitely see that being a common scenario. If that route is taken I think the Profile Editor shortcode should follow suit. Perhaps all `else` conditions for shortcodes are converted to an action. Existing content can be hooked in for backwards compatibility but can easily be removed without having to duplicate the whole template file (running in to this on most shortcodes with our site redesign) 

https://github.com/easydigitaldownloads/easy-digital-downloads/blob/master/templates/shortcode-profile-editor.php#L185-L186

Somewhat related #4943 which also helps shortcodes be modified without duplicating template files.

**Comment by SDavisMedia** [comment]
All the default messages are displaying on `issue/4922` as they should. Two notes, though:

* I can't get `edd_register_form_logged_in` to work with a custom function. It's not outputting anything. No idea why: https://cl.ly/3m3Q3V3R010K
* I think we should take this opportunity to give the paragraphs a class. The already logged in message has one. https://github.com/easydigitaldownloads/easy-digital-downloads/commit/b912d926785126333c354bd0cb40ea877ff3ca59#diff-51f5634c32065fe063f49ada8352b799R32 I think the rest should too since it would make sense to style these elements.

**Comment by sunnyratilal** [comment]
@SDavisMedia I addressed your second point [here](https://github.com/easydigitaldownloads/easy-digital-downloads/commit/a593ddaa869603be7efbf9a3b19d38f82a677363#diff-51f5634c32065fe063f49ada8352b799R19)

**Comment by sunnyratilal** [comment]
@SDavisMedia first point fixed [here](https://github.com/easydigitaldownloads/easy-digital-downloads/commit/7b6107200f8bd1da585a585d9ea24dd0674a92e1)

**Comment by SDavisMedia** [comment]
Both good now. Thanks. 👌

**Comment by sunnyratilal** [comment]
Merged into `release/2.8`

**Comment by cklosowski** [comment]
This has broken unit tests.

I'm reopening to fix it in `release/2.8` directly:
```
1) Tests_Shortcode::test_login_form
Failed asserting that '
	You are already logged in<p class="edd-logged-in"></p>
' contains "<p class="edd-logged-in">You are already logged in</p>".
```

We're using `_e` functions within an `echo` statement:
https://github.com/easydigitaldownloads/easy-digital-downloads/pull/5779/files#diff-51f5634c32065fe063f49ada8352b799R19
https://github.com/easydigitaldownloads/easy-digital-downloads/pull/5779/files#diff-51f5634c32065fe063f49ada8352b799R32
